### PR TITLE
Remove unused code (3/N)

### PR DESCRIPTION
**Pull request by est31**
Third PR in a series of PRs to remove unused/dead code from servo, powered by an (upcoming) tool of mine. Please take a look and tell me if you want to keep something.

* First PR: #23477
* Second PR: #23498

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [ ] These changes fix #___ (GitHub issue number if applicable)

- [ ] There are tests for these changes OR
- [x] These changes do not require tests because they only remove dead code

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/23499)

### CODE REVIEW

File: components/selectors/context.rs [hunk 1]
```diff
@@ -189,15 +189,6 @@ where
         }
     }
 
-    /// Override the quirks mode we're matching against.
-    ///
-    /// FIXME(emilio): This is a hack for XBL quirks-mode mismatches.
-    #[inline]
-    pub fn set_quirks_mode(&mut self, quirks_mode: QuirksMode) {
```
Comment (jdm) [inline components/selectors/context.rs]: Can't remove this; it's used by Gecko.

**Review by jdm — CHANGES_REQUESTED** [review]

File: components/selectors/context.rs (hunk 1 above)
Comment (est31) [reply components/selectors/context.rs]: Thanks for pointing this out, I'll leave it inside. The tool isn't perfect, it sometimes can't even detect when something is used by servo itself. You have to verify its output manually for false positives. I've ran some of the stuff through searchfox but I guessed I missed this one.

Comment (jdm) [reply components/selectors/context.rs]: Oh, that's a good point about SearchFox. You may actually be right that this is unused, but I would want @emilio to verify that.

Comment (emilio) [reply components/selectors/context.rs]: Yeah, this is used: https://searchfox.org/mozilla-central/rev/f8b11433159cbc9cc80500b3e579d767473fa539/servo/ports/geckolib/glue.rs#3734

**Comment by est31** [comment]
I've re-added the code examples in a way that doesn't touch the git blame. r? @jdm

**Comment by CYBAI** [comment]
@bors-servo retry

- #23290

**Pull request merged**